src/System.Net.Primitives/src/System/Net/CookieContainer.cs

```diff
@@ -0,0 +1,1062 @@
+                        // If expired, remove from container and don't add to the destination
+                        source.RemoveAt(idx);
+                        --_count;
+                        --idx;
+                    }
+                    else
+                    {
+                        // Add only if port does match to this request URI
+                        // or was not present in the original response.
+                        if (isPlainOnly && cookie.Variant != CookieVariant.Plain)
+                        {
+                            ; // Don't add
+                        }
+                        else if (cookie.PortList != null)
+                        {
+                            foreach (int p in cookie.PortList)
+                            {
+                                if (p == port)
+                                {
+                                    to_add = true;
+                                    break;
+                                }
+                            }
+                        }
+                        else
+                        {
+                            // It was implicit Port, always OK to add.
+                            to_add = true;
+                        }
+
+                        // Refuse to add a secure cookie into an 'unsecure' destination
+                        if (cookie.Secure && !isSecure)
+                        {
+                            to_add = false;
+                        }
+
+                        if (to_add)
+                        {
+                            // In 'source' are already orederd.
+                            // If two same cookies come from different 'source' then they
+                            // will follow (not replace) each other.
+                            destination.InternalAdd(cookie, false);
+                        }
+                    }
+                }
+            }
+        }
+
+        public string GetCookieHeader(Uri uri)
+        {
+            if (uri == null)
+            {
+                throw new ArgumentNullException("uri");
+            }
+            string dummy;
+            return GetCookieHeader(uri, out dummy);
+        }
+
+        internal string GetCookieHeader(Uri uri, out string optCookie2)
+        {
+            CookieCollection cookies = InternalGetCookies(uri);
+            string cookieString = String.Empty;
+            string delimiter = String.Empty;
+
+            foreach (Cookie cookie in cookies)
+            {
+                cookieString += delimiter + cookie.ToString();
+                delimiter = "; ";
+            }
+            optCookie2 = cookies.IsOtherVersionSeen ?
+                          (Cookie.SpecialAttributeLiteral +
+                           Cookie.VersionAttributeName +
+                           Cookie.EqualsLiteral +
+                           Cookie.MaxSupportedVersion.ToString(NumberFormatInfo.InvariantInfo)) : String.Empty;
```

Comment:
These all all constants, but the ToString here is going to force a string to be allocated for MaxSupportedVersion and then a string to be allocated for the concatenation.  Since MaxSupportedVersion is 1, could a MaxSupportedVersionLiteral const be added as "1" and used here?  Or if for some reason we must use ToString like this, could this whole string be cached in a readonly static string?